powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/model/ActivationContext.java

```diff
@@ -0,0 +1,254 @@
+     * Get activation ID.
+     * @return Activation ID.
+     */
+    public String getActivationId() {
+        return activationId;
+    }
+
+    /**
+     * Set activation name.
+     * @param activationName Activation name.
+     */
+    public void setActivationName(String activationName) {
+        this.activationName = activationName;
+    }
+
+    /**
+     * Get activation name.
+     * @return Activation name.
+     */
+    public String getActivationName() {
+        return activationName;
+    }
+
+    /**
+     * Set activation flags.
+     * @param activationFlags Activation flags.
+     */
+    public void setActivationFlags(List<String> activationFlags) {
+        this.activationFlags = activationFlags;
+    }
+
+    /**
+     * Get activation flags.
+     * @return Activation flags.
+     */
+    public List<String> getActivationFlags() {
+        return activationFlags;
+    }
+
+    /**
+     * Set activation status.
+     * @param activationStatus Activation status.
+     */
+    public void setActivationStatus(ActivationStatus activationStatus) {
+        this.activationStatus = activationStatus;
+    }
+
+    /**
+     * Get activation status.
+     * @return Activation status.
+     */
+    public ActivationStatus getActivationStatus() {
+        return activationStatus;
+    }
+
+    /**
+     * Set blocked reason.
+     * @param blockedReason Blocked reason.
+     */
+    public void setBlockedReason(String blockedReason) {
+        this.blockedReason = blockedReason;
+    }
+
+    /**
+     * Get blocked reason.
+     * @return Blocked reason.
+     */
+    public String getBlockedReason() {
+        return blockedReason;
+    }
+
+    /**
+     * Set application ID.
+     * @param applicationId Application ID.
+     */
+    public void setApplicationId(long applicationId) {
+        this.applicationId = applicationId;
+    }
+
+    /**
+     * Get application ID.
+     * @return Application ID.
+     */
+    public long getApplicationId() {
+        return applicationId;
+    }
+
+    /**
+     * Set user ID.
+     * @param userId User ID.
+     */
+    public void setUserId(String userId) {
+        this.userId = userId;
+    }
+
+    /**
+     * Get user ID.
+     * @return User ID.
+     */
+    public String getUserId() {
+        return userId;
+    }
+
+    /**
+     * Set version.
+     * @param version Version.
+     */
+    public void setVersion(long version) {
+        this.version = version;
+    }
+
+    /**
+     * Get version.
+     * @return Version.
+     */
+    public long getVersion() {
+        return version;
+    }
+
+    /**
+     * Set timestamp created.
+     * @param timestampCreated Timestamp created.
+     */
+    public void setTimestampCreated(XMLGregorianCalendar timestampCreated) {
+        this.timestampCreated = timestampCreated;
+    }
+
+    /**
+     * Get timestamp created.
+     * @return Timestamp created.
+     */
+    public XMLGregorianCalendar getTimestampCreated() {
```

Comment:
I am not sure about the data type - it is XML-related.

Reply:
This will eventually get resolved by abandoning SOAP completely. However, let's use `Instant` and use similar logic as we do for PAS related conversions...